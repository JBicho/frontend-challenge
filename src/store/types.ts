export interface ITranscription {
  id: number;
  voice: string;
  text: string;
}

export interface ITranscriptionState {
  transcriptionList: ITranscription[];
}

export interface IRootState {
  transcriptions: {
    list: ITranscription[];
  };
}
