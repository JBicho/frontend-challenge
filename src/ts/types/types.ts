export interface ITranscription {
  id: number;
  voice: string;
  text: string;
}
