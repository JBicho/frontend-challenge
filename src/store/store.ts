import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import { IRootState } from './types';
import { transcription } from './modules/transcriptions';

Vue.use(Vuex);

const store: StoreOptions<IRootState> = {
  state: {
    app: ''
  },
  modules: {
    transcription
  }
};

export default new Vuex.Store<IRootState>(store);
