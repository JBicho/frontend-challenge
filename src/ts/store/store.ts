import { createBrowserHistory } from "history";
import { routerMiddleware } from "react-router-redux";
import { applyMiddleware, combineReducers, compose, createStore } from "redux";
import createSagaMiddleware from "redux-saga";
import { all } from "redux-saga/effects";
import { transcriptionsReducer } from "./transcriptions/reducer";
import { transcriptionsSaga } from "./transcriptions/saga";
import { ITranscriptionsState } from "./transcriptions/state";

export interface IAppState {
  transcriptionsState: ITranscriptionsState;
}

const sagaMiddleware = createSagaMiddleware();

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const browserHistory = createBrowserHistory();

export const store = createStore(
  combineReducers<IAppState>({
    transcriptionsState: transcriptionsReducer
  }),
  composeEnhancers(
    applyMiddleware(sagaMiddleware, routerMiddleware(browserHistory))
  )
);

sagaMiddleware.run(function*() {
  yield all([transcriptionsSaga()]);
});
