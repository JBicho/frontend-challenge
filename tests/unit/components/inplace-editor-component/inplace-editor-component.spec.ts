import Vue from 'vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import InplaceEditor from '@/components/inplace-editor-component/inplace-editor-component.vue';
import Vuex from 'vuex';
import { ITranscription } from '@/store/types';

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuex);

describe('Add Row Component Tests', () => {
  let wrapper: any;
  let store: any;
  let actions: any;
  let state: any;
  let transcriptionInfo: ITranscription;

  beforeEach(() => {
    transcriptionInfo = {
      id: 1,
      voice: 'Test Voice',
      text: 'Test description'
    };

    actions = {
      editTranscription: jest.fn(() => true)
    };

    state = {
      transcriptionList: [],
      errors: []
    };

    store = new Vuex.Store({
      state: {},
      actions,
      modules: {
        transcription: {
          namespaced: true,
          state,
          actions
        }
      }
    });
  });

  test('Snapshot test', () => {
    const Constructor = Vue.extend(InplaceEditor);
    const vm = new Constructor().$mount();
    expect(vm.$el).toMatchSnapshot();
  });

  test('It should show text values when initialized with a transcription', () => {
    wrapper = shallowMount(InplaceEditor, {
      propsData: { transcriptionInfo }
    });

    expect(wrapper.find('.inplace-editor--voice').text()).toBe(
      transcriptionInfo.voice
    );
    expect(wrapper.find('.inplace-editor--text').text()).toBe(
      transcriptionInfo.text
    );
  });

  test('Should display input and textarea with transcription values alternately when the texts are clicked', () => {
    wrapper = shallowMount(InplaceEditor, {
      propsData: { transcriptionInfo }
    });

    wrapper.find('.inplace-editor--voice').trigger('click');

    expect(wrapper.find('input').element.value).toBe(transcriptionInfo.voice);

    wrapper.find('.inplace-editor--text').trigger('click');

    expect(wrapper.find('textarea').element.value).toBe(transcriptionInfo.text);
  });

  test('Should pass action to store on save button click', () => {
    wrapper = shallowMount(InplaceEditor, {
      propsData: { transcriptionInfo },
      store,
      localVue
    });

    wrapper.find('.inplace-editor--voice').trigger('click');

    wrapper.find('.inplace-editor--button').trigger('click');

    expect(actions.editTranscription).toHaveBeenCalled();

    wrapper.find('.inplace-editor--text').trigger('click');

    wrapper.find('.inplace-editor--button').trigger('click');

    expect(actions.editTranscription).toHaveBeenCalled();
  });
});
