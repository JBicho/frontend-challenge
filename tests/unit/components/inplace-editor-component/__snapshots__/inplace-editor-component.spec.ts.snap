// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Row Component Tests Snapshot test 1`] = `
<div
  class="item-info"
>
  <div
    class="inplace-editor"
  >
    <h3
      class="inplace-editor--voice"
    >
      Write your transcription title here
    </h3>
     
    <!---->
  </div>
   
  <div
    class="inplace-editor"
  >
    <p
      class="inplace-editor--text"
    >
      Write your transcription text here
    </p>
     
    <!---->
  </div>
</div>
`;
